app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,29 @@ def show
     @post = Post.find(params[:id])
     @post_comments = Comment.where(post_id: @post)
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.comments_counter = 0
+    @post.likes_counter = 0
```

Comment:
- [ ]  [OPTIONAL] In db level we have have set `comments_counter & likes_counter` as default 0, so we don't have to set here.